DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

cristina-sirbu opened the pull request:
The exciting and shiny new Hello World UI.

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Project clean up

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Added call to ACS and APS in homepage

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Added call to HelloWorld Backend Service

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Removed the top level folder

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Added gitignore and Contributing file back. Removed Dockerfile and changed ReadMe

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Removed package-lock.json

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Update app.config.json

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Removed test.ts

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Added copyright to source files

kconkas commented on LICENSE:
```diff
@@ -1,191 +1,21 @@
+MIT License
+
+Copyright (c) 2017 Denys Vuika
+
```
Why did we change the license for this project from Apache to MIT and why is this project's copyright owned by Denys Vuika?

cristina-sirbu replied:
This is the new license that gets generated using [Yeoman Generator Angular 2 Alfresco Application](https://github.com/Alfresco/generator-ng2-alfresco-app) and it states: ```The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.```

rgauss replied:
We should keep the license as Apache and copyright as Alfresco Software, Inc.

kconkas commented on src/app.config.json:
```diff
@@ -0,0 +1,69 @@
+    "$schema": "../node_modules/@alfresco/adf-core/app.config.schema.json",
+    "ecmHost": "http://localhost",
+    "bpmHost": "http://localhost",
+    "backendHost": "http://localhost:8080",
```
I find this naming of configuration variables rather confusing: we are naming variables `<something>Host`, yet they actually hold URLs.

cristina-sirbu replied:
I didn't name those for ACS and APS. They are generated. I created just the one for backend service and I tried to follow the pattern.

kconkas commented on src/app.config.json:
```diff
@@ -0,0 +1,69 @@
+    
+    "providers" : "ALL",
+    "application": {
+        "name": "Alfresco ADF Appplication"
```
Do we really want to stick with "Alfresco ADF Application" as our hello-world app name?

cristina-sirbu replied:
Forgot to change it. Thank you!

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Forgot to add pom.xml

kconkas reviewed (changes requested):
Overall excellent job. Minor changes may be appropriate (please see my comments).

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Added back alfresco-anaxes-hello-world-ui-tar.xml for the Bamboo build

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Changed the name of the App in app.config.json

kconkas reviewed (approved):
LGTM but let's discuss the issue of this project from now on by default being under MIT license and copyrighted by an individual Alfresco employee (unrelated to this project) before merging.

ehardon commented:
LGTM! 👍

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Changed the license back to Apache

rgauss commented on package.json:
```diff
@@ -1,60 +1,81 @@
+  "version": "2.6.0",
+  "blueprint": "Process and Content Services",
+  "description": "ACS APS ADF Application with Angular CLI",
+  "license": "MIT",
```
Probably need Apache here as well.

cristina-sirbu replied:
Thank you! Changed it in my last commit.

rgauss reviewed (changes requested):
Minor license cleanup

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Changed the License type in package.json

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Changed the app.config.json to be able to set the sso urls as env vars

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Update base href

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Update redirect uri

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Made the Redirect Uri a variable to be set at deploy time

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Changing back the redirectURI

Commit:
DEPLOY-530: Security vulnerability in Alfresco/alfresco-anaxes-hello-world-ui

Changed the urls to ACS and APS in order to be consistent with the current documentation
